client/client-lib/src/mint/backup.rs

```diff
@@ -216,5 +330,393 @@ impl EcashBackup {
+        let note_idx_ref = self.next_pending_note_idx.get_mut_or_default(amount);
+
+        let (note_issuance_request, blind_nonce) = NoteIssuanceRequest::new(
+            secp256k1::SECP256K1,
+            MintClient::new_note_secret_static(&self.secret, amount, *note_idx_ref),
+        );
+        assert!(self
+            .pending_nonces
+            .insert(
+                blind_nonce.0,
+                (note_issuance_request, *note_idx_ref, amount)
+            )
+            .is_none());
+
+        note_idx_ref.advance();
+    }
+
+    pub fn handle_input(&mut self, input: &MintInput) {
+        // We attempt to delete any nonce we see as spent, simple
+        for (_amt, note) in input.0.iter_items() {
+            self.spendable_note_by_nonce.remove(&note.0);
+        }
+    }
+
+    pub fn handle_output(&mut self, out_point: OutPoint, output: &MintOutput) {
+        // There is nothing preventing other users from creating valid transactions
+        // mining coins to our own blind nonce, possibly even racing with us.
+        // Including amount in blind nonce derivation helps us avoid accidentally using
+        // a nonce mined for as smaller amount, but it doesn't eliminate completely
+        // the possibility that we might use a note mined in a different transaction,
+        // that our original one.
+        // While it is harmless to us, as such duplicated blind nonces are effective as good
+        // the as the original ones (same amount), it breaks the assumption that all our
+        // blind nonces in an our output need to be in the pending pool. It forces us to be
+        // greedy no matter what and take what we can, and just report anything suspicious.
+        //
+        // found - all nonces that we found in the pool with the correct amount
+        // missing - all the nonces we have not found in the pool, either because they are not ours
+        //           or were consumed by a previous transaction using this nonce, or possibly gap
+        //           buffer was too small
+        // wrong - nonces that were ours but were mined to a wrong
+        let (found, missing, wrong) = output.0.iter_items().fold(
+            (vec![], vec![], vec![]),
+            |(mut found, mut missing, mut wrong), (amount_from_output, nonce)| {
+                match self.pending_nonces.get(&nonce.0).cloned() {
+                    Some((issuance_request, note_idx, pending_amount)) => {
+                        // the moment we see our blind nonce in the epoch history, correctly or incorrectly used,
+                        // we know that we must have used already
+                        self.observe_nonce_idx_being_used(pending_amount, note_idx);
+
+                        if pending_amount == amount_from_output {
+                            found.push((amount_from_output, (nonce.0, Some(issuance_request))));
+                            // replenish pending pool with the nonce of the same amount
```

Comment:
Is this a TODO?

Reply:
Leftoever. Replenishing pending nonces moved to `observe_nonce_idx_being_used`  Thanks for pointint it out.

Reply:
I assume it will be fixed in a follow-up